ios/TPSTwitterModule/TPSTwitterModule.m

```diff
@@ -0,0 +1,222 @@
+        NSString *paramsString = @"x_auth_mode=reverse_auth&";
+        //  Create the authorization header and attach to our request
+        NSData *bodyData = [paramsString dataUsingEncoding:NSUTF8StringEncoding];
+        NSString *authorizationHeader = OAuthorizationHeader(url, @"POST", bodyData, [Twitter sharedInstance].authConfig.consumerKey, [Twitter sharedInstance].authConfig.consumerSecret, nil, nil);
+        NSMutableURLRequest *request = [NSMutableURLRequest requestWithURL:url];
+        request.HTTPMethod = @"POST";
+        [request setValue:authorizationHeader forHTTPHeaderField:@"Authorization"];
+        request.HTTPBody = bodyData;
+        
+        [NSURLConnection sendAsynchronousRequest:request queue:[[NSOperationQueue alloc] init] completionHandler:^(NSURLResponse *response, NSData *data, NSError *connectionError) {
+            completion(data, connectionError);
+        }];
+    });
+}
+
+- (void)_step2WithAccount:(ACAccount *)account signature:(NSString *)signedReverseAuthSignature andHandler:(TWTAPIHandler)completion {
+    //TwitterKit must be used only from the main thread
+    dispatch_async(dispatch_get_main_queue(), ^{
+        NSDictionary *step2Params = @{@"x_reverse_auth_target": [Twitter sharedInstance].authConfig.consumerKey, @"x_reverse_auth_parameters": signedReverseAuthSignature};
+        NSURL *authTokenURL = [NSURL URLWithString:@"https://api.twitter.com/oauth/access_token"];
+        SLRequest *step2Request = [SLRequest requestForServiceType:SLServiceTypeTwitter requestMethod:SLRequestMethodPOST URL:authTokenURL parameters:step2Params];
+        step2Request.account = account;
+        [step2Request performRequestWithHandler:^(NSData *responseData, NSHTTPURLResponse *urlResponse, NSError *error) {
+            dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
+                completion(responseData, error);
+            });
+        }];
+    });
+}
+
+- (NSDictionary *)parseQueryString:(NSString *)string {
+    NSMutableDictionary *queryStringDictionary = [[NSMutableDictionary alloc] init];
+    NSArray *urlComponents = [string componentsSeparatedByString:@"&"];
+    
+    for (NSString *keyValuePair in urlComponents) {
+        NSArray *pairComponents = [keyValuePair componentsSeparatedByString:@"="];
+        NSString *key = [pairComponents objectAtIndex:0];
+        NSString *value = [pairComponents objectAtIndex:1];
+        
+        [queryStringDictionary setObject:value forKey:key];
+    }
+    
+    return queryStringDictionary;
+}
+
+#pragma mark - NSError
+
+- (NSError*)errorWithDescription:(NSString*)errorDescription {
+    NSDictionary *userInfo = @{
+                               NSLocalizedDescriptionKey:errorDescription,
+                               };
+    NSError *error = [NSError errorWithDomain:@"com.tipsi.twitter"
+                                         code:0
```

Comment:
Which codes you want use here? NSError use HTTP status codes, but I have no idea how use this code in custom errors.